src/data/fintype/card.lean

```diff
@@ -26,6 +26,22 @@ namespace fintype
 lemma card_eq_sum_ones {α} [fintype α] : fintype.card α = (finset.univ : finset α).sum (λ _, 1) :=
 finset.card_eq_sum_ones _
 
+section
+open finset
+
+variables {ι : Type*} [fintype ι] [decidable_eq ι]
+
+@[to_additive]
+lemma prod_extend_by_one [comm_monoid α] (s : finset ι) (f : ι → α) :
+  univ.prod (λ i, if i ∈ s then f i else 1) = s.prod f :=
```

Comment:
The following proof works:
```lean
by rw [← prod_filter, filter_mem_eq_inter, univ_inter]
```
if you add
```lean
@[simp] lemma univ_inter [decidable_eq α] (s : finset α) :
  univ ∩ s = s := ext' $ λ a, by simp

@[simp] lemma inter_univ [decidable_eq α] (s : finset α) :
  s ∩ univ = s :=
by rw [inter_comm, univ_inter]
```
to `finset/basic.lean`.

Reply:
Done, thanks